### PR TITLE
Add synonym support to rasa data.

### PR DESCRIPTION
**Pull request by jreeter**
**Proposed changes**:
- Add synonym support to load_rasa_data in converters.py.
- Synonyms to be described under "synonym_examples" as property of "rasa_nlu_data" e.g: 

```json
"rasa_nlu_data": {
    "synonym_examples": [
      {
        "value": "chinese",
        "synonyms": ["Chinese", "Chines", "chines"]
      }
    ],
    ...
}
```

**Status**:
- [x] ready for code review
- [x] documentation updated
- [x] changelog updated

Please let me know about any better approaches or naming conventions as I am new to the Python world.

### CODE REVIEW

File: data/examples/rasa/demo-rasa.json [hunk 1]
```diff
@@ -1,5 +1,11 @@
 {
   "rasa_nlu_data": {
+    "synonym_examples": [
```
Comment (tmbo) [inline data/examples/rasa/demo-rasa.json]: I'd rename it to `entity_synonyms`. I think that makes the use case clearer

Comment (jreeter) [inline data/examples/rasa/demo-rasa.json]: Sounds good @tmbo I'll get these changes in shortly.

**Comment by amn41** [comment]
As I understand it, if a user adds 'chinees' as a synonym, then **if** 'chinees' is recognised as a cuisine entity, it will resolve to the value 'chinese', which is great.

However, this doesn't in any way guarantee that 'chinees' actually would be picked up as a cuisine in the first place. 'NYC' versus 'the big apple' is probably a more obvious example. 

I'm worried that users will expect this to 'just work' and be disappointed. Maybe it would make sense to duplicate those entities with the synonyms swapped in to increase the chances they are recognised?

**Comment by jreeter** [comment]
@amn41 I actually kinda assumed that automatically happens during the training currently, is that not the case? If not, then the trainer should see an entity_synonyms entry into the rasa data files then as part of the training function automatically duplicate the training data with the proper synonyms. Am I off base here or is that what you're suggesting?

**Comment by tmbo** [comment]
@amn41 agreed.

@jreeter no, thats not happening yet.

**Comment by jreeter** [comment]
@tmbo I think I have covered all your requests. I can move forward with implementing what I described above per @amn41's point. I just wanted to make sure that what I described sounded like a good solution?

**Comment by amn41** [comment]
yes that sounds like the right thing generally. One question is whether we use this duplicates to also train the intents or not. I guess it's better to exclude them from the intent classification and only add these duplicates to the `entity_examples`. Otherwise if you add a bunch of synonyms you will potentially end up with very unbalanced classes.

**Comment by jreeter** [comment]
Ok, I can start with that thanks for the feedback @amn41.

**Comment by tmbo** [comment]
@jreeter did you already have time to look at this?

**Comment by znat** [comment]
Hey there, just wondering if there's any plan to merge this soon. Thanks

**Comment by tmbo** [comment]
@znat I agree, we should get this merged even without the data augmentation (which might be a bigger and separate project on its own). Nevertheless, the merge conflicts need to be resolved first.

**Comment by jreeter** [comment]
Merge conflicts have been fixed.

**Comment by tmbo** [comment]
looks good! just tested it, it's an issue with the ner extractor not with the synonyms. 

thanks for the work you have put into this :+1:

**Pull request merged**